src/presentation/cells/support.ts

```diff
@@ -1,31 +1,29 @@
-import { myAutoresizeOptions } from '../uiutils';
+import {autoresize, inputWidthUpdate, myAutoresizeOptions, triggerFocus} from '../uiutils';
 import { VNode } from 'snabbdom/vnode';
 import { VNodeChildElement } from 'snabbdom/h';
-import { ModelNode } from '../../datamodel/modelNode';
+import { ModelNode } from '../../datamodel';
 
 import { getDatamodelRoot } from '../../datamodel/registry';
 import { log } from '../../utils/misc';
+import _ from 'lodash';
 
 export function handleSelfDeletion(element: HTMLElement, modelNode: ModelNode): void {
-  if ($(element).closest('.represent-node').hasClass('deleting')) {
+  const closest = element.closest('.represent-node');
+  if (closest?.classList.contains('deleting')) {
     modelNode.deleteMe();
   } else {
-    $(element).closest('.represent-node').addClass('deleting');
+    closest?.classList.add('deleting');
   }
 }
 
 export function handleAddingElement(element: HTMLElement, modelNode: ModelNode): void {
   log('adding element', element, modelNode);
-  const parents = $(element).parents();
-
-  // First find the collection containing this node
-  const parentsArray = Array.from(parents);
-  const collectionIndex = parentsArray.findIndex((e) => $(e).hasClass('represent-collection'));
-  const parentsToConsider = parentsArray.slice(0, collectionIndex);
-  const nodesToConsider = $(parentsToConsider).filter('.represent-node');
-  const lastNode = nodesToConsider.last();
```

Comment:
Was this all part totally useless?

Reply:
Yes, it iterated over the list of parents of a single element, filtering it according to classes. But an element only has a single parent. jQuery parents() makes sense only if you're invoking it on a list of elements – it's like map(#parent)